core/http/src/header/mod.rs

```diff
@@ -3,11 +3,11 @@ mod known_media_types;
 mod media_type;
 mod content_type;
 mod accept;
-mod header;
+mod header_impl;
```

Comment:
Why? Let's revert this.

Reply:
All the module rename was to fix the warning [module_inception](https://rust-lang.github.io/rust-clippy/master/index.html#module_inception). I did it because all of their content are exported by the parent module so renaming them is a minor change. I would allow the lint if it had changed the public API or more than one file. I knew you might not like that change, this is why I've put it in a separate commit (I did forget to move some non-trivial fixes in this commit because there were lots of fixes).

The goal of this pull request is to suggest solutions to fix the lint instead of drowning the source code in `allow(...)`. To put you in context, one of my employers asked us to enable all GCC's warnings and to consider them as errors -- including the pedantic ones. All the code looked like this one  :
```
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "{Some lint}"
 // ...
#pragma GCC diagnostic pop
```
This is why I did my best to offer solutions that don't need to add `allow(...)` to fix warnings.

Since you own this repository, I don't mind adding `allow(...)` for the solution you don't like. As I mention earlier, I'll only add one per ignored lint per file to keep the code clean (without the justifying comment). I'll let you decide which `allow(..)` should be applied on what level.